### PR TITLE
[JS] Topic: Create a "Hello World" tutorial

### LINKED ISSUE #69: JS: Create a "Hello World" tutorial
# Create a "Hello World" tutorial
Create a simple Hello World tutorial for JavaScript.

## Directory
```
Topics/1.Hello_World
```

## Description
- The most significant code-block is going to be.
  ```js
  console.log('Hello World!');
  ```
- Mention how to run the Hello World script in an HTML page
  and where to see the output.
- Mention how to run the same script inside Node.js.
- How can we print to the HTML document instead of the console in
  the web browser? Hint: Using `document.write`.

### PR DESCRIPTION
**Pull request by AbhinandanAdhikari**
<hr>

## Description 📜

Created a tutorial to display Hello World in JavaScript.
Fixes #69

<hr>

## Type of change 📝

- [ ] Audio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Video (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev (Flutter, Java, Kotlin)
- [ ] Backend Dev (Java, .NET, Python)
- [ ] C/CPP
- [ ] Competitive Programming
- [ ] Cyber Security
- [ ] DSA
- [ ] Database
- [ ] Datascience with Python
- [ ] Datascience with R
- [ ] Frontend Dev (HTML, CSS, JS)
- [ ] Frontend Dev (React, Angular, Vue)
- [ ] Golang
- [ ] Interview Prep
- [ ] Java
- [x] JavaScript
- [ ] MERN
- [ ] Machine Learning
- [ ] Open Source
- [ ] Python
- [ ] Rust
- [ ] Statistics
- [ ] UI/UX

<hr>
 
## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

File: Web_Development/Javascript/Topics/1.Hello_World/README.md [hunk 1]
```diff
@@ -1 +1,110 @@
...
+  res.statusCode = 200;
+  res.setHeader('Content-Type', 'text/plain');
+  res.end('Hello, World!');
+});
+
+server.listen(port, hostname, () => {
+  console.log(`Server running at http://${hostname}:${port}/`);
+});
+```
+3. Then open your command prompt in that folder and type ```node .\script.js```
```
Comment (paramsiddharth) [inline Web_Development/Javascript/Topics/1.Hello_World/README.md:78]: "`.\`" (backslashes) aren't platform-agnostic and specific to Windows (and mostly PowerShell on Windows).

Make it `node script.js`.

**Review by paramsiddharth — CHANGES_REQUESTED** [review]
The rest looks amazing to me! Well done, Abhinandan. Go ahead with the changes and push to the same branch. The PR will be updated and I'll review it further.

File: Web_Development/Javascript/Topics/1.Hello_World/README.md (hunk 1 above)
Comment (AbhinandanAdhikari) [reply Web_Development/Javascript/Topics/1.Hello_World/README.md]: Yes Sir! I've updated it.

Comment (AbhinandanAdhikari) [reply Web_Development/Javascript/Topics/1.Hello_World/README.md]: @paramsiddharth  sir can you please review the changes made

**Review by paramsiddharth — APPROVED** [review]

**Comment by paramsiddharth** [comment]
@Creatoon @dakshsethi Could you please review the PR, and request any changes if needed?

**Comment by Creatoon** [comment]
@paramsiddharth This is the best Hello world example I have ever seen, but what I think is - the beginner who will start coding with JS will get a fear after looking at so many ways, what I can suggest is keep it simple and short, just console log one, because hello world program is just for giving a flavour of a language to newbies.
What do you think ?

**Comment by Creatoon** [comment]
It will more better if you just write console.log("Hello World")
@paramsiddharth what you say

File: Web_Development/Javascript/Topics/1.Hello_World/README.md [hunk 2]
```diff
@@ -1 +1,110 @@
...
+</body>
+<script>
+    alert('Hello, World!');
+</script>
+</html>
+```
+__Explanation:__
+
+The Output will be generated in an alert box of your browser. 
+
```
Comment (dakshsethi) [inline Web_Development/Javascript/Topics/1.Hello_World/README.md:50]: Here we can add a screenshot of the alert box of the browser with `Hello World!` printed inside it.

File: Web_Development/Javascript/Topics/1.Hello_World/README.md [hunk 3]
```diff
@@ -1 +1,110 @@
...
+```document.write``` is used when you want the display the content to the HTML document.
+
+__Source Code:__
+```javascript
+document.write('Hello, World!');
+```
+__Output:__
+```
+Hello, World!
+```
```
Comment (dakshsethi) [inline Web_Development/Javascript/Topics/1.Hello_World/README.md:100]: Here also we can attach a screenshot image of the HTML page.

**Review by dakshsethi — CHANGES_REQUESTED** [review]

**Comment by paramsiddharth** [comment]
> @paramsiddharth This is the best Hello world example I have ever seen, but what I think is - the beginner who will start coding with JS will get a fear after looking at so many ways, what I can suggest is keep it simple and short, just console log one, because hello world program is just for giving a flavour of a language to newbies.
> What do you think ?

Actually, I think it is better to show all the different ways JavaScript can be used in the Hello World tutorial. :) One developer might be using JavaScript for one thing, another for something else. Besides it was mentioned in the issue that way itself, that's why he did it like that. I think that should be alright!

**Comment by dakshsethi** [comment]
> > @paramsiddharth This is the best Hello world example I have ever seen, but what I think is - the beginner who will start coding with JS will get a fear after looking at so many ways, what I can suggest is keep it simple and short, just console log one, because hello world program is just for giving a flavour of a language to newbies.
> > What do you think ?
> 
> Actually, I think it is better to show all the different ways JavaScript can be used in the Hello World tutorial. :) One developer might be using JavaScript for one thing, another for something else. Besides it was mentioned in the issue that way itself, that's why he did it like that. I think that should be alright!

But @paramsiddharth don't you think we can at least remove the node.js `Hello World!` version from here and add it in the Node.js section at later stage?

File: Web_Development/Javascript/Topics/1.Hello_World/README.md (hunk 3 above)
Comment (AbhinandanAdhikari) [reply Web_Development/Javascript/Topics/1.Hello_World/README.md:102]: @Creatoon  @dakshsethi  Sir, I've attached the images. Please Review my Pull Request.

**Review by dakshsethi — APPROVED** [review]
Yes both the images are great.

**Comment by paramsiddharth** [comment]
> > > @paramsiddharth This is the best Hello world example I have ever seen, but what I think is - the beginner who will start coding with JS will get a fear after looking at so many ways, what I can suggest is keep it simple and short, just console log one, because hello world program is just for giving a flavour of a language to newbies.
> > > What do you think ?
> > 
> > 
> > Actually, I think it is better to show all the different ways JavaScript can be used in the Hello World tutorial. :) One developer might be using JavaScript for one thing, another for something else. Besides it was mentioned in the issue that way itself, that's why he did it like that. I think that should be alright!
> 
> But @paramsiddharth don't you think we can at least remove the node.js `Hello World!` version from here and add it in the Node.js section at later stage?

Not at all. I think that the Hello World file must highlight the fact that JavaScript is not merely a front-end programming language anymore and has significance as a DOM-tool, front-end scripting language, and a back-end scripting language. One may choose it to operate on one particular stack or multiple parts of an application.

**Comment by paramsiddharth** [comment]
@Creatoon Please review.

**Comment by AbhinandanAdhikari** [comment]
@Creatoon  Sir, Please review.

**Review by Creatoon — APPROVED** [review]

**Review by s4-2 — APPROVED** [review]

**Comment by s4-2** [comment]
@AbhinandanAdhikari  Nice job :). The code looks simple and any newbie can understand this easily.
Congratulation, keep it up.

**Comment by AbhinandanAdhikari** [comment]
> @AbhinandanAdhikari Nice job :). The code looks simple and any newbie can understand this easily.
> Congratulation, keep it up.

@s4-2 Thankyou Ma'am

**Comment by AbhinandanAdhikari** [comment]
@paramsiddharth Sir, is it ready to merge?

**Review by dakshsethi — APPROVED** [review]

**Pull request merged**